pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1473,6 +1469,28 @@ default 'raise'
                  self.nanosecond / 3600.0 / 1e+9
                 ) / 24.0)
 
+    def strftime(self, format: str) -> str:
+        """
+        Constructs datetime style `format` string from Timestamp.
+
+        See `datetime <https://docs.python.org/3/library/datetime\
+.html#strftime-and-strptime-format-codes>`_ module for all available directives.
+
+        Parameters
+        ----------
+        format : str
+            String of formatting directives
+
+        Returns
+        -------
+        str
+            String representation of Timestamp
+        """
+        if self.nanosecond and '%f' in format:
```

Comment:
Is there a reason for this check and the one on L649? Can there just be one check done here?